docs/how-to/uc-image-creation.rst

```diff
@@ -5,11 +5,141 @@ How to create a real-time Ubuntu Core image
 
     This guide assumes access to features available exclusively to `dedicated Snap Store`_ users.
 
-The :doc:`../how-to/uc-boot-parameters` describes the method for dynamically configuring the system for real-time processing.
-The configuration is an iterative process that is best done together with the expected workload. 
-Once satisfied with the configurations, it is time to prepare the operating system for going into production.
+Canonical builds and publishes Ubuntu Core images for a range of `supported platforms`_.
+However, these images include the generic Linux kernel. 
+In order to run Ubuntu Core with the real-time kernel, we need to build it ourselves. 
 
-This guide shows how to statically set the desired Kernel command-line options for the Ubuntu Core system.
+This guide shows how to build and Ubuntu Core image with the real-time kernel.
+We start by showing how to :ref:`build a vanilla image <ubuntu-core-image>` which is excellent for testing and tuning, and then illustrate how to :ref:`make a custom image <custom-ubuntu-core-image>` with production-ready kernel configurations.
+
+
+.. _ubuntu-core-image:
+
+Create an image with the real-time kernel
+-----------------------------------------
+
+To do this we need to describe the image content and then use a tool to build it.
+
+.. _model-assertion:
+
+Create the model assertion
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The model assertion is a digitally signed document that describes the content of the Ubuntu Core image.
+Read the `model assertion`_ documentation before continuing.
+
+Below is an example model assertion in YAML, describing a `core22` Ubuntu Core
+image:
+
+.. literalinclude :: uc-image-creation/model.json
+   :language: json
+
+Inside an empty directory, create a file named ``model.json`` with the above content.
+
+Change the following:
+
+- ``authority-id``, ``brand-id`` to your developer ID, since this is custom model. Use ``snapcraft whoami`` command to get your developer ID.
+- ``timestamp`` to a RFC3339 formatted string, within you signing key's validity. Use ``date -Iseconds --utc`` command to use the current time.
+- ``store`` to your dedicated Snap Store ID.
+- ``model`` to something representative of your model.
+
+The ``snaps`` array is a list of snaps that get included in the image.
+In that list, the ``realtime-kernel`` snap contains the realtime Linux kernel.
+Here you can add any other snaps, including for example your real-time applications.
+
+Next, we need to sign the model assertion.
+Refer to the guide on `signing model assertion`_ for details on how to sign the model assertion. 
+
+Here are the needed steps:
+
+1) Create and register a key
+
+
+.. code-block:: shell
+
+    snapcraft create-key realtime-ubuntu
+    snapcraft register-key realtime-ubuntu
+
```

Comment:
```suggestion

.. warning::

    Choose the name and password for the key carefully, as once registered, it's not possible to change or remove the key.
```

Reply:
> We don't know when, but once that happens,

Not sure if that will ever happen. If they didn't implement key deletion at the time they implemented key registration, there must be another reason for it.

Reply:
This is a presumption on someone else's product. A bit of research shows that the snapcraft CLI used to support revocation, and there are locally available revocation certificates that can be used for each key. 

https://bugs.launchpad.net/snapstore-server/+bug/1669471

```console
$ tree /home/farshid/.snap/gnupg/
/home/farshid/.snap/gnupg/
├── openpgp-revocs.d
│   ├── 32C84846EF48EF6BD8F97CC63DDF5FF093D7F9B2.rev
│   ├── 5D2F0F60249592D1A49CB80BA8C4F62D7912B49E.rev
│   ├── 9142937D8F532836CC5D23E713E7CFC9CCED927E.rev
│   └── 93F56C15D58577B69DADB02DC92BA06F835B494F.rev
├── private-keys-v1.d
│   ├── 2F138999CDDB4A3C9730D681624AF2534DCFD7E8.key
│   ├── 8060443C480D86672B610D4B70E3BB6D662E784F.key
│   └── E2814AF986655128FC204D0FF044B1790EB315C7.key
├── pubring.kbx
├── pubring.kbx~
├── sshcontrol
└── trustdb.gpg
```